### ISSUE TITLE
jethro/public shows rosters for many many week (all populated?) rather than 'roster week default'

### ISSUE THREAD

**Issue by s4069b** [tbar0970/jethro-pmm#449]
not sure if this is a bug or if I have broken something? Or if it is expected behaviour?

Public rosters are visible till 30th December (? because tasks have been filled till that date?) rather than the 'roster weeks default' in <admin> <system configuration>.   I've tried changing that number but it makes no difference. 

I would much prefer public only show a limited number of weeks... because beyond the next month or so things are subject to change.

**Comment by tbar0970** [comment]
This is the intended behaviour at this point.

The "roster weeks default" setting controls the default date range selected in the "view roster" page in the logged-in interface - just as the starting point.  But the public page shows all weeks that have a service saved.

We could add a new setting "public roster weeks" or similar.

**Comment by tbar0970** [comment]
What I can imagine being useful is the concept of "publishing" a roster up to a certain date.  But this introduces a fair bit of extra complexity for both the code and the users.

**Comment by s4069b** [comment]
Thanks for getting back. Glad to know I have not broken something (I thought I might have when I had to delete and re-insert Easter Friday after setting the date incorrectly). 

a 'public roster weeks' setting would be fantastic. 

Personally I prefer a readily updated roster to a 'published' roster. The public roster view allows everyone to check whenever they want.

**Comment by tbar0970** [comment]
> Personally I prefer a readily updated roster to a 'published' roster. The public roster view allows everyone to check the 'current' roster whenever they want (rather than a static/published one).

Yes, absolutely.  By "published" I mean "made visible to non-editors" rather than "made static".  So once you had finalised your assignments for term 2 you could make that exact period of the roster visible publicly.  But it wouldn't be too easy - just a dream!

**Comment by tbar0970** [comment]
+1
